LoongArch-ELF-ABI-EN.adoc

```diff
@@ -0,0 +1,206 @@
+= LoongArch ELF ABI specification
+Loongson Technology Corporation Limited
+V1.00
+
+:encoding: UTF-8
+:frame: top
+:grid: rows
+:valign: middle
+
+== Register Convention
+.Integer Register Convention
+[width="80%",options="header",cols="2m,2m,^5,^3"]
+|====================================================================
+|Name     |Alias        |Meaning             |Preserved across calls
+|$r0      |$zero        |Constant zero       |--
+|$r1      |$ra          |Return address      |No
+|$r2      |$tp          |Thread pointer      |--(Unallocatable)
+|$r3      |$sp          |Stack pointer       |Yes
+|$r4-$r11 |$a0-$a7      |Argument registers  |No
+|$r4-$r5  |$v0-$v1      |Return value        |No
```

Comment:
It's unnecessary to give `$r4-$r5` alternative names just to maintain some shade of MIPS similarity; AArch64 and RISC-V both reuse `$a0-a1` as return value registers but they make do without the aliases. Plus this is putting burden on learners; I struggle to remember this even today.

Reply:
Reusing argument registers for returns values is a good idea. ppc32/ppc64 do this as well.